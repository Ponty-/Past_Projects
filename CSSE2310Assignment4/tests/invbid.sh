#!/bin/bash

if [ $# != 4 ]
then
    echo "Usage: $0 binarydir port resultdir correct_dir" 
    exit 1
fi



bindir=$1
port=$2
res=$3
right=$4
name=$0

expected=0

if [ ! -f $bindir/client499 ]
then
    echo "$bindir/client499 is missing"
    exit 1
fi

if [ ! -f $bindir/serv499 ]
then
    echo "$bindir/serv499 is missing"
    exit 1
fi



$bindir/serv499 $port .. 1.deck &
spid=$!
$bindir/client499 A game $port > $res/$name.a.out 2>$res/$name.a.err < iamoves &
pid1=$!
$bindir/client499 B game $port > $res/$name.b.out 2>$res/$name.b.err < ibmoves &
pid2=$!
$bindir/client499 C game $port > $res/$name.c.out 2>$res/$name.c.err < icmoves &
pid3=$!
$bindir/client499 D game $port > $res/$name.d.out 2>$res/$name.d.err < idmoves &
pid4=$!

wait $pid1 $pid2 $pid3 $pid4 2> /dev/null
kill $spid 2> /dev/null
wait $spid 2> /dev/null

#if [ $result != $expected ]
#then
#    echo "Expected $expected got $result"
#    exit 1
#fi

for v in a b c d
do

diff -q $res/$name.$v.out $right/$name.$v.out > /dev/null
if [ $? != 0 ]
then
   echo "Stdout mismatch for client $v"
   echo "diff $res/$name.$v.out $right/$name.$v.out"
   exit 2
fi

diff -q $res/$name.$v.err $right/$name.$v.err > /dev/null
if [ $? != 0 ]
then
   echo "Stderr mismatch for client $v"
   echo "diff $res/$name.$v.err $right/$name.$v.err"
   exit 3
fi

done

echo OK
exit 0

