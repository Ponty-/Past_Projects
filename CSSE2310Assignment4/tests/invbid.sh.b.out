Info: ..
Info: Team1: A, C
Info: Team2: B, D
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
Info: A bids 7D
[7D] - Bid (or pass)> Info: C bids 9H
Info: C plays AD
Info: D plays AS
Info: A plays KD
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
[D] play> Info: C won
Info: C plays QD
Info: D plays KS
Info: A plays TD
S: T 6
C: A T 6 2
D: J 9 6
H: K 9 4
[D] play> Info: C won
Info: C plays 8D
Info: D plays JS
Info: A plays 5D
S: T 6
C: A T 6 2
D: J 9
H: K 9 4
[D] play> Info: B won
S: T 6
C: A T 6 2
D: J
H: K 9 4
Lead> Info: C plays 9S
Info: D plays 8S
Info: A plays 7S
Info: C won
Info: C plays KC
Info: D plays JC
Info: A plays 3C
S: T
C: A T 6 2
D: J
H: K 9 4
[C] play> Info: C won
Info: C plays JH
Info: D plays 3H
Info: A plays 2H
S: T
C: A T 6
D: J
H: K 9 4
[H] play> Info: C won
Info: C plays 4S
Info: D plays 3S
Info: A plays QS
S: T
C: A T 6
D: J
H: K 9
[S] play> Info: A won
Info: A plays 7D
S:
C: A T 6
D: J
H: K 9
[D] play> Info: C plays 4D
Info: D plays 5S
Info: B won
S:
C: A T 6
D:
H: K 9
Lead> Info: C plays 9C
Info: D plays 8C
Info: A plays 7C
Info: B won
S:
C: T 6
D:
H: K 9
Lead> Info: C plays 4C
Info: D plays 5C
Info: A plays QC
Info: A won
Info: A plays 6H
S:
C: 6
D:
H: K 9
[H] play> Info: C plays 8H
Info: D plays 7H
Info: B won
S:
C: 6
D:
H: K
Lead> Info: C plays 2D
Info: D plays 2S
Info: A plays AH
Info: A won
Info: A plays TH
S:
C:
D:
H: K
[H] play> Info: C plays 5H
Info: D plays QH
Info: B won
Info: Team 1=-300, Team 2=0
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
Info: A bids 9H
Info: A plays KD
S: T 6
C: A T 6 2
D: J 9 6 3
H: K 9 4
[D] play> Info: C plays AD
Info: D plays 2S
Info: C won
Info: C plays JH
Info: D plays QH
Info: A plays 2H
S: T 6
C: A T 6 2
D: J 9 6
H: K 9 4
[H] play> Info: D won
Info: D plays AS
Info: A plays QS
S: T 6
C: A T 6 2
D: J 9 6
H: K 9
[S] play> Info: C plays 9S
Info: D won
Info: D plays KS
Info: A plays 7S
S: 6
C: A T 6 2
D: J 9 6
H: K 9
[S] play> Info: C plays 4S
Info: D won
Info: D plays JS
Info: A plays QC
S:
C: A T 6 2
D: J 9 6
H: K 9
[S] play> Info: C plays KC
Info: D won
Info: D plays 8S
Info: A plays 3C
S:
C: T 6 2
D: J 9 6
H: K 9
[S] play> Info: C plays 9C
Info: D won
Info: D plays 5S
Info: A plays 7C
S:
C: 6 2
D: J 9 6
H: K 9
[S] play> Info: C plays 4C
Info: D won
Info: D plays 3S
Info: A plays 5D
S:
C: 2
D: J 9 6
H: K 9
[S] play> Info: C plays QD
Info: D won
Info: D plays JC
Info: A plays TD
S:
C:
D: J 9 6
H: K 9
[C] play> Info: C plays 8D
Info: D won
Info: D plays 8C
Info: A plays 7D
S:
C:
D: 9 6
H: K 9
[C] play> Info: C plays 4D
Info: D won
Info: D plays 5C
Info: A plays AH
S:
C:
D: 9
H: K 9
[C] play> Info: C plays 2D
Info: A won
Info: A plays 6H
S:
C:
D:
H: K 9
[H] play> Info: C plays 8H
Info: D plays 7H
Info: B won
S:
C:
D:
H: 9
Lead> Info: C plays 5H
Info: D plays 3H
Info: A plays TH
Info: A won
Info: Team 1=-600, Team 2=0
Info: Winner is Team 2
